### ISSUE TITLE
IllegalThreadStateException

### ISSUE THREAD

**Issue by AgiMaulana** [sendbird/SendBird-Android#66]
```
2019-12-27 14:25:07.846 23687-23910/App.Package E/AndroidRuntime: FATAL EXCEPTION: Thread-124
    Process: App.Package  PID: 23687
    java.lang.IllegalThreadStateException
        at java.lang.Thread.start(Thread.java:724)
        at com.sendbird.syncmanager.MessageSynchronizer.startPrev(MessageSynchronizer.java:196)
        at com.sendbird.syncmanager.MessageSynchronizer.access$800(MessageSynchronizer.java:11)
        at com.sendbird.syncmanager.MessageSynchronizer$1$1$1$1$1.onCompleted(MessageSynchronizer.java:132)
        at com.sendbird.syncmanager.MessageChunkContainer$3$1.onCompleted(MessageChunkContainer.java:99)
        at com.sendbird.syncmanager.MessageChunkContainer$4.onCompleted(MessageChunkContainer.java:127)
        at com.sendbird.syncmanager.DatabaseController$17.run(DatabaseController.java:1111)
        at java.lang.Thread.run(Thread.java:764)
```

Our app shows message descendingly from the oldest to the most recent, shows the most recent messages in the bottom.

In the explanation below, the "first" means older, the last means "most recent".

on our app, when the user scrolls into first 100 messages the app loads 100 of previous messages since the last message shown on the UI. When the user scrolls into the last 50 items the app loads 100 of next messages, the most recent since the most recent message shown on the UI. But sometimes, the above error occurred and made our app crashed.

**Comment by BC-Chung** [comment]
Thanks for the report.
I will let our dev team to look at this issue.

Best regards
BC Chung
Solutions Engineer 
SendBird

**Comment by AgiMaulana** [comment]
This is the SendBird's dependencies we were used

    ```
// SendBird
    implementation 'com.sendbird.sdk:sendbird-android-sdk:3.0.111'
    // SendBird SyncManager
    implementation 'com.sendbird.sdk:sendbird-syncmanager:1.1.17'
```

**Comment by AgiMaulana** [comment]
Thank you for following up this issue

**Comment by BC-Chung** [comment]
Thanks. I will let you know the investigation result from our dev team.

**Comment by BC-Chung** [comment]
Hello AgiMaulana

Here is an update.
As far as we know this issue isn't a blocker nor happens frequently.
So, our dev team is targeting to apply the fix for this issue on SyncManager 2.0 release.
Currently, we are targeting to release this SyncManager 2.0 before end of 2020 Q1.
Thanks!

Best regards
BC Chung
Solutions Engineer
SendBird

**Comment by AgiMaulana** [comment]
Thank you @BC-Chung .
We wish version 2.0 will have better performance. Good luck, SendBird.

**Comment by BC-Chung** [comment]
Hello AgiMaulana

I got to know that this issue has been resolved in SyncManager v1.1.18 as well.
Hope this helps!
https://github.com/sendbird/sendbird-syncmanager-android/blob/master/CHANGELOG.md#v1118-feb-5-2020

Best regards
BC Chung
Solutions Engineer
SendBird
